### PR TITLE
Adopt Airbnb swift format package for formatting and linting

### PR DESCRIPTION
**Pull request by calda**
## Change summary

This PR adopts the new Airbnb swift format package plugin from https://github.com/airbnb/swift/pull/168. This lets us remove the dependency on `mint`, and remove the direct dependencies on SwiftFormat / SwiftLint (now managed by the format plugin).

## How was it tested?
- [x] Wrote automated tests
- [ ] Built and ran on the iOS simulator
- [ ] Built and ran on a device

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -39,3 +39,8 @@ let package = Package(
     .testTarget(name: "EpoxyTests", dependencies: ["Epoxy", "Quick", "Nimble"]),
     .testTarget(name: "PerformanceTests", dependencies: ["EpoxyCore"]),
   ])
+
+#if swift(>=5.6)
+// Add the Airbnb Swift formatting plugin if possible
+package.dependencies.append(.package(url: "https://github.com/airbnb/swift", .branch("cal--error-improvements")))
```
Comment (calda) [inline Package.swift:45]: Waiting for https://github.com/airbnb/swift/pull/173 to merge, then I'll update this to use `.branch("master")`.

**Review by erichoracek — APPROVED** [review]
Thanks, looks great!

File: Rakefile [hunk 2]
```diff
@@ -35,8 +35,7 @@ namespace :lint do
 
   desc 'Lints swift files'
   task :swift do
-    sh 'mint run SwiftLint lint Sources Example --config script/lint/swiftlint.yml --strict'
-    sh 'mint run SwiftFormat Sources Example --config script/lint/airbnb.swiftformat --lint'
+    sh 'swift package --allow-writing-to-package-directory format --lint --exclude Tests'
```
[the remark this answers is not in the document]
Comment (calda) [reply Rakefile:38]: There's a file in the tests folder that causes SwiftLint to error out / crash 😫

[the remark this answers is not in the document]
Comment (calda) [reply Rakefile:38]: It doesn't give us any info about the crash, and the file is like 1000 lines long, so I have no clue what the problem is.

**Comment by calda** [comment]
hmmmm this is a problem -- the `bundle exec rake lint:swift` job is failing in CI with a `error: unknownError(exitCode: 10)` error that I can't reproduce locally. That type of mysterious failure happens to me locally when I try to lint the `Tests` folder but not the other folders. Really not sure what's going wrong with SwiftLint here.

**Comment by calda** [comment]
Good news is that top-of-tree master for SwiftLint doesn't crash mysteriously, let me try that

File: Rakefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (calda) [reply Rakefile:38]: Adopting the latest SwiftLint commit seems to fix the error I was encountering -- although there are also like 90 lint failures (from `print`s and force unwraps) that we would need to resolve before we can start linting that folder.

**Comment by calda** [comment]
This is ready to merge now that it points to the `master` branch of the style guide repo 👍🏻

**Comment by erichoracek** [comment]
Awesome, thank you Cal!

**Pull request merged**